frontend-react/src/components/ManagePublicKey/ManagePublicKeyUpload.tsx

```diff
@@ -12,55 +12,78 @@ import {
 export interface ManagePublicKeyUploadProps {
     onPublicKeySubmit: (e: React.FormEvent<HTMLFormElement>) => void;
     onFileChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
+    onBack: () => void;
+    hasBack: boolean;
+    hasPublicKey: boolean;
     file: File | null;
 }
 
 export default function ManagePublicKeyUpload({
     onPublicKeySubmit,
     onFileChange,
+    onBack,
+    hasBack,
+    hasPublicKey,
     file,
```

Comment:
I feel like we can simplify this to a "publicKey" prop that can be `boolean | File`. if publicKey == true then show already configured, else show the form. Parent component would pass something like `publicKey={hasPublicKey ?? file}` in component initialization. What do you think @stephenkao ?

Reply:
I agree, and I feel like something similar could be done with `onBack`/`hasBack` -- we could make it just not pass `onBack` if the component shouldn't show a back button 🤷

Reply:
Oh yeah, good catch!